catalogue_graph/src/ingestor/transformers/base_transformer.py

```diff
@@ -0,0 +1,84 @@
+import json
+import os
+from collections.abc import Generator
+from typing import IO, Any
+
+import boto3
+import polars as pl
+import smart_open
+from pydantic import BaseModel
+
+from ingestor.extractors.base_extractor import GraphBaseExtractor
+from ingestor.steps.ingestor_indexer import IngestorIndexerObject
+from utils.types import IngestorLoadFormat
+
+
+class ElasticsearchBaseTransformer:
+    def __init__(self, start_offset: int, end_index: int, is_local: bool) -> None:
+        self.source: GraphBaseExtractor = GraphBaseExtractor(
+            start_offset, end_index, is_local
+        )
+
+    def transform_document(self, raw_document: Any) -> BaseModel | None:
+        """
+        Accepts raw data representing the item to be transformed and returns an indexable document as a Pydantic model.
+        """
+        raise NotImplementedError(
+            "Each Elasticsearch transformer must implement a `transform_document` method."
+        )
+
+    def _load_to_file(self, file: IO, load_format: IngestorLoadFormat) -> int:
+        es_documents = list(self.stream_es_documents())
+        if len(es_documents) == 0:
+            raise ValueError("No documents to write.")
+
+        if load_format == "parquet":
+            df = pl.DataFrame(es_documents, infer_schema_length=None)
```

Comment:
I did wonder about whether we should hand a schema in here again, but we are essentially enforcing a schema with pydantic types ... I wonder if there is a way to make that _more_ enforced in the conversion and reading from parquet, [something like this.](https://github.com/simw/pydantic-to-pyarrow) 

I don't think this is necessary now, but it would be nice to have some guarantee beyond the model validation after loading that we are accessing the correct types.